subprojects/core/src/main/java/org/gradle/internal/cleanup/BuildOutputCleanupCache.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cleanup;
+
+import java.io.File;
+
+public interface BuildOutputCleanupCache {
+
+    String CACHE_DISPLAY_NAME = "Build Output Cleanup Cache";
+
+    /**
+     * Cleans stale output files from previous builds. Stale files are left behind under the following conditions:
+     *
+     * - The task history is deleted and one or more input files are removed.
+     * - The build upgrades or downgrades to a different Gradle version, wasn't executed with that version before and one or more input files are removed.
```

Comment:
These are the conditions under which stale files/directories are left behind as indicated by the previous sentence. I'll move it to a new line to make it more clear.